### ISSUE TITLE
Cadastro completo logo de cara

### ISSUE THREAD

**Issue by apzeiro** [deivisonarthur/OSC-Magento-Brasil-6-Pro#223]
Pessoal, gostaria de saber se é possível configurar para que o Magento solicite todos os dados do cliente no cadastro inicial?
O que andou a acontecendo é que o cliente faz o cadastro, que basicamente é nome e-mail e senha e daí já quer comprar, mas fica faltando preencher endereço CPF etc...

Existe uma forma de pegar esse cadastro completo logo de cara?

**Comment by philspbr** [comment]
Claro, é só adicionar esses campos no seu template de registro. Abs

**Comment by apzeiro** [comment]
@philspbr , poderia me dar uma dica de como fazer isso? Nunca fiz e não sei nem por onde começar.

**Comment by vissoci** [comment]
Achei uma solução interessante. Existe um módulo para o Magento que resolve esse problema. 
[Username Support - Login | Register | Checkout - by Diglin]

Corrigindo o link para o projeto no Github: https://github.com/diglin/Diglin_Username

**Comment by hansbonini** [comment]
O Módulo já tem o cadastro completo, é só seguir as instruções no Readme do projeto.

**Comment by apzeiro** [comment]
@hansbonini creio que esteja equivocado, não possui essa opção no módulo. Se tiver, cole o trecho aqui por gentileza. 

O cadastro simplificado (sem endereço) é nativo do magento.
Creio que esteja confundindo com o cadastro do checkout. Lá sim é possível fazer alguns ajustes com o OSC. 
O cadastro que eu me refiro é o "customer/account/create/"
A extensão sugerida acima habilita o endereço para esse cadastro.

**Comment by hansbonini** [comment]
@apzeiro como disse, é só seguir as instruções do README.md
``pre
Tutoriais e Observações

A - Habilitando os campos de endereços

A1 - Visite a página: Admin : Sistema > Configuração > Clientes > Configuração > "Opções de Nome e Endereço" >

A2 - Aletera para:

    Número de Linhas p/ Endereço:4
    Exibir Data de Nascimento:Opcional
    Exibir CPF/CNPJ:Obrigatorio
    Exibir Sexo:Opcional

Ou seja, ficará assim:

    street1 = endereço
    street2 = numero
    street3 = complemento
    street4 = bairro

B - Habilite o Estado como obrigatório:

B1 - Visite a página: Admin : Sistema > Configuração "Aba Geral " > Opção de estado "Estado é necessário para" >

B2 - Selecione: Brasil

C - Habilite o taxvat e coloque como obrigatório:

C1 - Visite a página: Admin : Sistema > Configuração > Clientes > Configuração > "Opções ao Criar Nova Conta"

C2 - Selecione: Exibir CPF/CNPJ no Frontend: Sim
``

**Comment by apzeiro** [comment]
@hansbonini , essas instruções funcionam apenas para o checkout e não para o cadastro prévio como mencionado. O cadastro inicial simplificado é nativo do magento.  
Creio que haja confusão. Eu me refiro ao cadastro que a pessoa faz sem comprar.
Faça o teste em qualquer loja acessando: 
url-da-sua-loja/customer/account/create/

O bom é que o módulo indicado resolve essa questão.

**Comment by hansbonini** [comment]
@apzeiro sim... ambos são modificados e ficam iguais. O problema deve estar relacionado ao seu template, é só trocar pelo template rwd e conferir.

**Comment by apzeiro** [comment]
@hansbonini bom saber então! Vou testar pra ver e confere a informação.
Não testei no RWD, mas em todos os templates de todas as lojas que eu administro apresentam o mesmo problema.
Obrigado.

**Comment by hansbonini** [comment]
@apzeiro é que a maioria dos templates é desenvolvido para o cenário internacional, então os campos de cadastro não sofrem alterações como aqui. Geralmente esses templates precisam de modificações para trabalhar corretamente com o plugin. Qualquer coisa estamos ai.

**Comment by apzeiro** [comment]
no RWD também não funcionou. Creio que não seja possível realmente.

**Comment by vissoci** [comment]
Usei novamente o mesmo módulo em outra loja e resolve bem essa situação. Fica a dica do módulo:
Magento 1: https://github.com/diglin/Diglin_Username
Magento 2: https://github.com/diglin/Diglin_Username2
